terra/executor/celery/executor.py

```diff
@@ -251,126 +250,56 @@ def __init__(self, service_info):
+      sender.main_log_handler = StreamHandler(stream=sender._log_file)
+      sender.root_logger.addHandler(sender.main_log_handler)
 
-      super(CeleryExecutor, CeleryExecutor).configure_logger(sender, **kwargs)
-    else:
-      assert False, 'unknown zone: ' + settings.terra.zone
-
-# from https://docs.python.org/3/howto/logging-cookbook.html
-class LogRecordStreamHandler(socketserver.StreamRequestHandler):
-  """Handler for a streaming logging request.
-
-  This basically logs the record using whatever logging policy is
-  configured locally.
-  """
-
-  def handle(self):
-    """
-    Handle multiple requests - each expected to be a 4-byte length,
-    followed by the LogRecord in pickle format. Logs the record
-    according to whatever policy is configured locally.
-    """
-    while True:
-      chunk = self.connection.recv(4)
-      if len(chunk) < 4:
-        break
-      slen = struct.unpack('>L', chunk)[0]
-      chunk = self.connection.recv(slen)
-      while len(chunk) < slen:
-        chunk = chunk + self.connection.recv(slen - len(chunk))
-      obj = self.unPickle(chunk)
-      record = logging.makeLogRecord(obj)
-      self.handleLogRecord(record)
-
-  def unPickle(self, data):
-    return pickle.loads(data)
-
-  def handleLogRecord(self, record):
-    # if a name is specified, we use the named logger rather than the one
-    # implied by the record.
-    if self.server.logname is not None:
-      name = self.server.logname
-    else:
-      name = record.name
-    logger = terra.logger.getLogger(name)
-    # N.B. EVERY record gets logged. This is because Logger.handle
-    # is normally called AFTER logger-level filtering. If you want
-    # to do filtering, do it at the client end to save wasting
-    # cycles and network bandwidth!
-    logger.handle(record)
-
-class LogRecordSocketReceiver(socketserver.ThreadingTCPServer):
-  """
-  Simple TCP socket-based logging receiver suitable for testing.
-  """
-
-  allow_reuse_address = True
-
-  def __init__(self, host='localhost',
-               port=logging.handlers.DEFAULT_TCP_LOGGING_PORT,
-               handler=LogRecordStreamHandler):
-    socketserver.ThreadingTCPServer.__init__(self, (host, port), handler)
-    self.abort = 0
-    self.timeout = 1
-    self.logname = None
-
-  def serve_until_stopped(self):
-    import select
-    abort = 0
-    print('SGR - STARTING LISTNER')
-    while not abort:
-      rd, wr, ex = select.select([self.socket.fileno()],
-                                  [], [],
-                                  self.timeout)
-      if rd:
-        print('SGR - RD')
-        self.handle_request()
-      abort = self.abort
+  @staticmethod
+  def reconfigure_logger(sender, pre_run_task=False,
+                         post_settings_context=False, **kwargs):
+    # if settings.terra.zone == 'runner' or
+    if settings.terra.zone == 'task':
+      print("SGR - reconfigure task logging", kwargs)
+
+      if pre_run_task:
```

Comment:
Do you need to remove the NullHandler that was added in `confgure`?

Reply:
Good catch